test/case/meta/wait.py

```diff
@@ -19,6 +22,26 @@ def ll6ping(node):
     return None
 
 
+def is_reachable(node, env):
+    neigh = ll6ping(node)
+    if not neigh:
+        return False
+
+    password = env.ptop.get_password(node)
+    if env.transport is not None:
+        if env.transport == "netconf":
+            return netconf.netconf_syn(neigh)
+        elif env.transport == "restconf":
+            return restconf.restconf_reachable(neigh, password)
+        else:
+            raise Exception(f"Unsupported transport {env.transport}!")
+
+    netconf_reachable = netconf.netconf_syn(neigh)
+    restconf_reachable = restconf.restconf_reachable(neigh, password)
+
+    return netconf_reachable and restconf_reachable
```

Comment:
If this is moved to `util.py`, couldn't we reuse it at the end of `wait_boot()` to get rid of the code duplication?

I.e. replace that with

```python
until(lambda: is_reachable(...), attempts=300)
```

Reply:
Put it as future work in the commit message, but I can have a look.